tests/acceptance/blockchain/interfaces/test_handle_multiversion_contracts.py

```diff
@@ -0,0 +1,99 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from pathlib import Path
+
+from nucypher.blockchain.eth.sol.compile.config import ALLOWED_PATHS
+
+from nucypher.blockchain.eth.interfaces import BlockchainDeployerInterface, BlockchainInterfaceFactory
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+from nucypher.crypto.powers import TransactingPower
+from tests.constants import INSECURE_DEVELOPMENT_PASSWORD
+
+
+def test_deployer_interface_multiversion_contract():
+
+    # Prepare compiler
+    base_dir = Path(__file__).parent / 'contracts' / 'multiversion'
+    v1_dir, v2_dir = base_dir / 'v1', base_dir / 'v2'
+
+    # I am a contract administrator and I an compiling a new updated version of an existing contract...
+    # Represents "Manually hardcoding" a new permitted compile path in compile.py
+    # and new source directory on BlockchainDeployerInterface.SOURCES.
+    ALLOWED_PATHS.append(base_dir)
+    BlockchainDeployerInterface.SOURCES = (v1_dir, v2_dir)
```

Comment:
Heh, I also was once turned off by setting, bit it's a totally acceptable API IMO
 @jMyles  helped me warm up to it. Especially since this is in a test with a big fat comment explaining why like there is here.